crates/forge/src/gas_report.rs

```diff
@@ -205,27 +194,20 @@ impl GasReport {
                     let functions = contract
                         .functions
                         .iter()
-                        .map(|(fname, sigs)| {
-                            // If there is only one signature, display the gas info directly.
-                            let function_value = if sigs.len() == 1 {
-                                format_gas_info(sigs.values().next().unwrap())
-                            } else {
-                                // If there are multiple signatures, e.g. overloads like:
-                                // - `foo(uint256)`
-                                // - `foo(int256)`
-                                // display the gas info as a map with the signature as the key.
-                                let signatures = sigs
-                                    .iter()
-                                    .map(|(sig, gas_info)| {
-                                        let display_name = sig.replace(':', "");
-                                        (display_name, format_gas_info(gas_info))
-                                    })
-                                    .collect::<BTreeMap<_, _>>();
-
-                                json!(signatures)
-                            };
-
-                            (fname.to_string(), function_value)
+                        .flat_map(|(_, sigs)| {
+                            sigs.iter().map(|(sig, gas_info)| {
+                                let display_name = sig.replace(':', "");
+                                (
+                                    display_name,
+                                    json!({
+                                        "calls": gas_info.calls,
+                                        "min": gas_info.min,
+                                        "mean": gas_info.mean,
+                                        "median": gas_info.median,
+                                        "max": gas_info.max,
+                                    }),
```

Comment:
can gas_info impl Serialize? Otherwise I would prefer a separate struct rather than using the json macro

Reply:
Fixed, yeah we already do - we can just return `gas_info` here